cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/ObjectStores.java

```diff
@@ -32,14 +30,39 @@
  * Utility for describing {@link ObjectStore} and similar datasets within application configuration.
  */
 public final class ObjectStores {
-  private static final Gson GSON = new GsonBuilder()
-    .registerTypeAdapter(Schema.class, new SchemaTypeAdapter())
-    .create();
+  private static final Gson GSON = new Gson();
 
   private ObjectStores() {}
 
   /**
-   * Adds {@link ObjectStore} dataset to be created at application deploy if not exists.
+   * Adds a {@link ObjectMappedTable} dataset to be created at application deploy if it does not exist.
+   *
+   * @param configurer application configurer
+   * @param datasetName dataset name
+   * @param type type of objects to be stored in {@link ObjectMappedTable}
+   * @param props any additional dataset properties
+   * @throws UnsupportedTypeException
+   */
+  public static void createObjectMappedTable(ApplicationConfigurer configurer,
+                                             String datasetName, Type type, DatasetProperties props)
+    throws UnsupportedTypeException {
+
+    configurer.createDataset(datasetName, ObjectMappedTable.class, objectStoreProperties(type, props));
```

Comment:
So ObjectMappedTable uses the exact same properties as ObjectStore? But will that always be the case? Or should we have objectMappedTableProperties(...) for this?